### ISSUE TITLE
Dropout noise in regression

### ISSUE THREAD

**Issue by qazmichaelgw** [pytorch/pytorch#1207]
When I use the dropout layer in torch.nn.Dropout(). I found that it has much noise when predicting very simple cosine and the sine function. 

![normal](https://cloud.githubusercontent.com/assets/8761378/24785734/d38cf44e-1b21-11e7-94a5-280ef6864453.png)
![dropout_sgd](https://cloud.githubusercontent.com/assets/8761378/24785735/d38d44e4-1b21-11e7-9b08-9670476a6078.png)
![dropout_0 5](https://cloud.githubusercontent.com/assets/8761378/24785736/d38fb0a8-1b21-11e7-8d53-6254eb6ac53f.png)

While I use keras in both theano backend or tensorlflow backend, they can predict them perfectly. I wonder is there any problem in the dropout layer?
Thanks a lot.

**Comment by angusturner** [comment]
Are you remembering to disable dropout at test / inference time? Try calling `.train(mode=False)` on your model.

**Comment by soumith** [comment]
this certainly looks like you didn't put dropout in eval mode with `.train(False)` or `.eval()`
In the future, post questions at https://discuss.pytorch.org
